### PR TITLE
Story/quad 15

### PR DESCRIPTION
**Pull request by v1kd**
Adds Archive and Unarchive links on the workspace details page

### CODE REVIEW

**Comment by jdamerow** [comment]
pleae resolve conflicts

**Comment by v1kd** [comment]
Resolved conflicts. Please review

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/workspace/ListWSManager.java [hunk 1]
```diff
@@ -325,4 +326,10 @@ public boolean getDeactiveStatus(String workspaceId) throws QuadrigaStorageExcep
         return false;
     }
 
+    @Override
+    @Transactional
+    public boolean getArchiveState(String workspaceId) throws QuadrigaStorageException {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/workspace/ListWSManager.java]: please rename to "isWorkspaceArchived"

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java [hunk 2]
```diff
@@ -256,4 +288,38 @@ public ModelAndView unarchiveWorkspace(@Validated @ModelAttribute("workspaceform
...
+
+		// add redirect attributes
+		redirectAttributes.addFlashAttribute("show_success_alert", true);
+		redirectAttributes.addFlashAttribute("success_alert_msg",
+				"The workspace has been successfully unarchived.");
+
+		return "redirect:/auth/workbench/workspace/workspacedetails/"
+				+ workspaceId;
+	}
+
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java:321]: please use correct style guide (only 4 spaces)

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ListWSController.java [hunk 3]
```diff
@@ -335,6 +335,7 @@ public String getWorkspaceDetails(@PathVariable("workspaceid") String workspacei
 		
 		//Including a condition to check if the workspace is not deactive. If the workspace is deactive adding attribute to make delete button disabled
 		model.addAttribute("isDeactivated", wsManager.getDeactiveStatus(workspaceid));
+        model.addAttribute("isArchived", wsManager.getArchiveState(workspaceid));
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ListWSController.java]: formatting seems off

**Comment by jdamerow** [comment]
ok to test

File: Quadriga/src/main/webapp/WEB-INF/views/workspace/workspacedetails.jsp [hunk 4]
```diff
@@ -589,8 +599,106 @@
...
+                                        return false;
+                                    },
+                                    "No" : function() {
+                                        $(this).dialog('close');
+                                        return false;
+                                    }
+                                }
+                            });
+                }
+			</script> <c:choose>
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/workspace/workspacedetails.jsp]: Is there a reason you include this choose here again?

**Comment by jdamerow** [comment]
see comments

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java (hunk 2 above)
Comment (v1kd) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java:321]: The code right now uses tab space as indentation. I fixed the indentation

File: Quadriga/src/main/webapp/WEB-INF/views/workspace/workspacedetails.jsp (hunk 4 above)
Comment (v1kd) [inline Quadriga/src/main/webapp/WEB-INF/views/workspace/workspacedetails.jsp]: The choose is used for other check for some other functionality. I formatted the code.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ListWSController.java (hunk 3 above)
Comment (v1kd) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ListWSController.java]: Fixed the formatting. My IDE uses 4 spaces as the indentation. I changes the settings

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/workspace/ListWSManager.java (hunk 1 above)
Comment (v1kd) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/workspace/ListWSManager.java]: Done

**Comment by v1kd** [comment]
Refactored the code as per comments. Please review and merge.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ListWSController.java (hunk 3 above)
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ListWSController.java]: actually, we use 4 spaces as well (default in eclipse is 6 spaces though), but we haven't been adopting strict formatting guidelines until now. For new classes you write please use 4 spaces. Thanks.

**Comment by jdamerow** [comment]
ok to test

**Pull request merged**